pkg/media/samplebuilder/samplebuilder.go

```diff
@@ -0,0 +1,95 @@
+package samplebuilder
+
+import (
+	"github.com/pions/webrtc/pkg/media"
+	"github.com/pions/webrtc/pkg/rtp"
+)
+
+// SampleBuilder contains all packets
+// maxLate determines how long we should wait until we get a valid RTCSample
+// The larger the value the less packet loss you will see, but higher latency
+type SampleBuilder struct {
+	maxLate uint16
+	buffer  [65536]*rtp.Packet
+
+	// Last seqnum that has been added to buffer
+	lastPush uint16
+
+	// Last seqnum that has been successfully popped
+	// -1 means if no pop has occurred
+	lastPopSeq       int32
```

Comment:
I would probably keep lastPopSeq and all datatypes that operate on seqnums in uint16 rather than casting, and encode -1 somewhere else

Reply:
@kc5nra updated PR! thanks for the review